rotkehlchen/data_import/importers/bitstamp.py

```diff
@@ -0,0 +1,159 @@
+        - ValueError
+        """
+        timestamp = ts_sec_to_ms(deserialize_timestamp_from_date(
+            date=csv_row['Datetime'],
+            formatstr=timestamp_format,
+            location='Bitstamp',
+        ))
+
+        amount, amount_symbol = csv_row['Amount'].split(' ')
+        event_identifier = f'{BITSTAMP_EVENT_PREFIX}_{uuid4().hex}'
+
+        if csv_row['Type'] == 'Market':
+            value_amount, value_symbol = csv_row['Value'].split(' ')
+            fee_amount, fee_symbol = csv_row['Fee'].split(' ')
+            trade_type = TradeType.deserialize(csv_row['Sub Type'])
+
+            # if trade_type buy
+            bought_amount = deserialize_asset_amount(amount)
+            sold_amount = deserialize_asset_amount(value_amount)
+            bought_currency = asset_from_bitstamp(amount_symbol)
+            sold_currency = asset_from_bitstamp(value_symbol)
+
+            if trade_type == TradeType.SELL:
+                bought_amount, sold_amount = sold_amount, bought_amount
+                sold_currency, bought_currency = bought_currency, sold_currency
+
+            fee_amount = deserialize_fee(fee_amount)
+            fee_currency = asset_from_bitstamp(fee_symbol)
+            spend_trade_event = HistoryEvent(
+                event_identifier=event_identifier,
+                sequence_index=0,
+                timestamp=timestamp,
+                location=Location.BITSTAMP,
+                asset=sold_currency,
+                balance=Balance(
+                    amount=sold_amount,
+                    usd_value=ZERO,
+                ),
+                notes=f'Spend {sold_amount} {sold_currency} as the result of a trade on Bitstamp',
+                event_type=HistoryEventType.TRADE,
+                event_subtype=HistoryEventSubType.SPEND,
+            )
+            receive_trade_event = HistoryEvent(
+                event_identifier=event_identifier,
+                sequence_index=1,
+                timestamp=timestamp,
+                location=Location.BITSTAMP,
+                asset=bought_currency,
+                balance=Balance(
+                    amount=bought_amount,
+                    usd_value=ZERO,
+                ),
+                notes=f'Receive {bought_amount} {bought_currency} as the result of a trade on Bitstamp',  # noqa: E501
+                event_type=HistoryEventType.TRADE,
+                event_subtype=HistoryEventSubType.RECEIVE,
+            )
+            fee_event = HistoryEvent(
+                event_identifier=event_identifier,
+                sequence_index=2,
+                timestamp=timestamp,
+                location=Location.BITSTAMP,
+                asset=fee_currency,
+                balance=Balance(
+                    amount=fee_amount,
+                    usd_value=ZERO,
+                ),
+                notes=f'Fee of {fee_amount} {fee_currency} as the result of a trade on Bitstamp',
+                event_type=HistoryEventType.TRADE,
```

Comment:
If you check https://github.com/rotki/rotki/blob/develop/rotkehlchen/accounting/constants.py you will see that the combination of `type:TRADE, subtype:FEE` is not a valid/known combination.

This will result in the event as being invalid. The fee event is shown only when the type is SPEND and subtype FEE. So the event in this case won't appear in the UI, which is the exact reason we had to do https://github.com/rotki/rotki/pull/6553 for.

I think this may be a miscalculation on our side. And that perhaps we should just:

1. Add TRADE/FEE as a possible combination with `EventCategory.FEE` as result.
2. Replace SPEND/FEE with TRADE/FEE where needed, where it's just the 3rd part of a trade.

What do you think @yabir? Also do you think there is a nice way to make sure no part of our code produces invalid combinations? Otherwise we will end up finding this out when a user reports it

Reply:
I would add Trade/Fee there. For the decoded events we have this list of expected events and (I have to double check if it is there) but there should be a piece of code that during tests check that the produced value for each event is in the known combinations for that decoder. We should do something similar for the imports

Reply:
https://github.com/rotki/rotki/blob/fc3d4973db9c1274de4490f44ef4926ab7c4acb0/rotkehlchen/chain/evm/decoding/decoder.py#L961

Reply:
should I change it to SPEND/FEE pending when TRADE/FEE is added to valid combinations @LefterisJP

Reply:
No. Add Trade/FEE to valid combinations. Don't push changes when there is a pending review. It runs thetests for nothing